### PR TITLE
NO-JIRA Add AllClassesTest

### PR DESCRIPTION
**Pull request by brusdev**

### CODE REVIEW

File: tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/AllClassesTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      Arrays.sort(targetConstructors, (c1, c2) -> c2.getParameterCount() - c1.getParameterCount());
+      for (Constructor targetConstructor : targetConstructors) {
+         List<Object> initArgs = new ArrayList<>();
+         Parameter[] constructorParameters = targetConstructor.getParameters();
+
+         for (Parameter constructorParameter : constructorParameters) {
+            Object initArg = null;
+            if (constructorParameter.getType().isAssignableFrom(byte.class)) {
+               initArg = RandomUtil.randomByte();
+            } else if (constructorParameter.getType().isAssignableFrom(byte[].class)) {
```
Comment (clebertsuconic) [inline tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/AllClassesTest.java:104]: nice!!!!!!

Comment (clebertsuconic) [reply tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/AllClassesTest.java:104]: the randoms on the constructors... nice!!!

File: tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/AllClassesTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+   @Test
+   public void testToString() {
+      Object targetInstance = newInstance(targetClass);
+      Assume.assumeTrue("cannot create " + targetClass.getName(), targetInstance != null);
+
+      try {
+         String targetOutput = targetInstance.toString();
+         log.debug("targetOutput: " + targetOutput == null ? "null" : targetOutput);
+      } catch (NullPointerException | UnsupportedOperationException ignore) {
```
Comment (clebertsuconic) [inline tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/AllClassesTest.java:88]: Can't you just not call the default constructors?

a toString() from a well defined constructor should not throw any exceptions here.

perhaps we could fix whatever issues come after this?

Comment (brusdev) [reply tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/AllClassesTest.java:88]: The `newInstance` try to call any constructor of the target classes starting from the one with more parameters to the one with less parameters.

**Comment by brusdev** [comment]
This PR is replaced by #3854 that fixes toString methods throwing exceptions too.

**Pull request closed without merging**